driver-core/src/main/com/mongodb/internal/operation/AsyncCommandBatchCursor.java

```diff
@@ -95,17 +91,11 @@ class AsyncCommandBatchCursor<T> implements AsyncAggregateResponseBatchCursor<T>
         this.firstBatchEmpty = commandCursorResult.getResults().isEmpty();
 
         AsyncConnection connectionToPin = null;
-        boolean releaseServerAndResources = false;
-        if (limitReached()) {
-            releaseServerAndResources = true;
-        } else if (connectionDescription.getServerType() == ServerType.LOAD_BALANCER) {
+        if (connectionDescription.getServerType() == ServerType.LOAD_BALANCER) {
```

Comment:
Nit: Now that this is a single `if`, it can become a simple assignment with a ternary expression.

Reply:
Done